sdc/tests/test_dataframe.py

```diff
@@ -945,6 +945,39 @@ def test_impl(n):
         n = 11
         pd.testing.assert_frame_equal(hpat_func(n), test_impl(n))
 
+    def test_df_head_unbox(self):
+        def test_impl(df, n):
+            return df.head(n)
+        sdc_func = sdc.jit(test_impl)
+        for n in [-3, 0, 3, 5, None]:
+            for idx in [[3, 4, 2, 6, 1], None]:
+                df = pd.DataFrame({"FLOAT": [3.2, 4.4, 7.0, 3.3, 1.0],
+                                   "INT": [3, 4, 1, 0, 222],
+                                   "STRING": ['a', 'dd', 'c', '12', 'ddf']}, index=idx)
+                with self.subTest(n=n, index=idx):
+                    pd.testing.assert_frame_equal(sdc_func(df, n), test_impl(df, n))
+
+    def test_df_head(self):
+        def get_func(n):
+            def impl(a):
+                return a.head(n)
+
+            return impl
```

Comment:
It''s interesting why do you need implementation generator?

Reply:
This is necessary to test n which is getting outside the jit region